echidna/core/smear.py

```diff
@@ -292,6 +371,94 @@ def weight_gaussian_energy_spectra(self, true_spectrum, num_sigma=5.):
+                        sigma = self.get_energy_sigma(mean_energy)
+                        entries = float(true_spectrum._data[energy_bin,
+                                                            radial_bin,
+                                                            time_bin])
+                        if entries == 0:
+                            continue  # Bin Empty
+                        lower_bin = self.floor_to_bin(mean_energy-num_sigma*sigma,
+                                                      energy_step)+0.5*energy_step
+                        upper_bin = self.ceil_to_bin(mean_energy+num_sigma*sigma,
+                                                     energy_step)-0.5*energy_step
+                        if upper_bin > true_spectrum._energy_high:
+                            upper_bin = true_spectrum._energy_high-0.5*energy_step
+                        if lower_bin < true_spectrum._energy_low:
+                            lower_bin = true_spectrum._energy_low+0.5*energy_step
+                        weights = []
+                        for energy in np.arange(lower_bin, upper_bin, energy_step):
+                            weights.append(self.calc_gaussian(energy,
+                                                              mean_energy,
+                                                              sigma))
+                        i = 0
+                        tot_weight = np.array(weights).sum()
+                        for energy in np.arange(lower_bin, upper_bin, energy_step):
+                            try:
+                                smeared_chunk.fill(
+                                    energy, mean_radius, mean_time,
+                                    entries*weights[i]/tot_weight)
+                            except ValueError:
+                                # Occurs when smeared energy is outside bin range
+                                print "Warning: Smeared energy out of bounds. Skipping."
+                                continue
+                            i += 1
+            out_q.put(smeared_chunk) 
+            
+        smeared_spectrum = spectra.Spectra(
+            true_spectrum._name+str(self._light_yield)+"_light_yield",
+            true_spectrum._num_decays)
+        chunksize =  math.ceil(true_spectrum._time_bins/float(nProcs))
+        procs = []
+        out_q = Queue()
+        for i in range(nProcs):
+            lower = chunksize*i
+            upper = chunksize*(i+1)
+            if upper > true_spectrum._time_bins:
+                upper = true_spectrum._time_bins
+            p = multiprocessing.Process(
+            target=Worker,
+            args=(true_spectrum,lower,upper,num_sigma,out_q
+            ))
+            procs.append(p)
+            p.start()
+
+        for i in range(nProcs):
+            smeared_chunk = out_q.get()
+            smeared_spectrum.add(smeared_chunk)
+    
+        for p in procs:
+            p.join()
```

Comment:
This chunk of code from `smeared_spectrum = ...` looks like it is pretty much the same each time. Could this be turned into a function?